apps/io-services-cms-webapp/src/publicator/request-publication-handler.ts

```diff
@@ -1,34 +1,54 @@
-import { ServiceLifecycle, ServicePublication } from "@io-services-cms/models";
+import { Queue, ServicePublication } from "@io-services-cms/models";
 import { readableReport } from "@pagopa/ts-commons/lib/reporters";
 import * as E from "fp-ts/lib/Either";
 import * as TE from "fp-ts/lib/TaskEither";
 import { flow, pipe } from "fp-ts/lib/function";
 import { Json } from "io-ts-types";
+import { Context } from "@azure/functions";
 import { withJsonInput } from "../lib/azure/misc";
 
 const parseIncomingMessage = (
   queueItem: Json
-): E.Either<Error, ServiceLifecycle.definitions.Service> =>
+): E.Either<Error, Queue.RequestPublicationItem> =>
   pipe(
     queueItem,
-    ServiceLifecycle.definitions.Service.decode,
+    Queue.RequestPublicationItem.decode,
     E.mapLeft(flow(readableReport, (_) => new Error(_)))
   );
 
+export const handleQueueItem = (
+  context: Context,
+  queueItem: Json,
+  fsmPublicationClient: ServicePublication.FsmClient
+) =>
+  pipe(
+    queueItem,
+    parseIncomingMessage,
+    TE.fromEither,
+    TE.mapLeft((_) => new Error("Error while parsing incoming message")),
+    TE.chainW((service) =>
+      service.autoPublish
+        ? fsmPublicationClient.publish(service.id, {
+            data: {
+              id: service.id,
+              data: service.data,
+            },
+          })
+        : fsmPublicationClient.override(service.id, {
+            data: {
+              id: service.id,
+              data: service.data,
+            },
+          })
+    ),
+    TE.getOrElse((e) => {
+      throw e;
```

Comment:
I guess this `throw` causes a retry in the function handler. You probably want to tell transient errors from permament ones (ie. if the format of the payload is wrong, causing n-retries here would consume costly cycles without a reason).

Reply:
We will handle this issue with new pull request in the near future

Reply:
in these cases add a comment with a TODO and a reference to the JIRA ID  of the issues